src/core/server/server.ts

```diff
@@ -158,27 +158,31 @@ export class Server {
     );
   }
 
-  private registerCoreContext(coreSetup: {
-    http: InternalHttpServiceSetup;
-    elasticsearch: ElasticsearchServiceSetup;
-    savedObjects: SavedObjectsServiceSetup;
-  }) {
+  private registerCoreContext(
+    coreSetup: InternalCoreSetup,
+    savedObjects: SavedObjectsServiceSetup
+  ) {
     coreSetup.http.registerRouteHandlerContext(
       coreId,
       'core',
       async (context, req): Promise<RequestHandlerContext['core']> => {
         const adminClient = await coreSetup.elasticsearch.adminClient$.pipe(take(1)).toPromise();
         const dataClient = await coreSetup.elasticsearch.dataClient$.pipe(take(1)).toPromise();
+        const savedObjectsClient = savedObjects.clientProvider.getClient(req);
+
         return {
           savedObjects: {
             // Note: the client provider doesn't support new ES clients
             // emitted from adminClient$
-            client: coreSetup.savedObjects.clientProvider.getClient(req),
+            client: savedObjectsClient,
           },
           elasticsearch: {
             adminClient: adminClient.asScoped(req),
             dataClient: dataClient.asScoped(req),
           },
+          uiSettings: {
+            client: coreSetup.uiSettings.asScopedToClient(savedObjectsClient),
```

Comment:
Note: There are several reasons to pass `savedObjectsClient` here:
- Consistency. All context services use the same version of the client.
- There is no way for uiSettings to declare a dependency on the saved object service at the moment. The current dependency graph is `uiSettings` -- used by --> `legacy` -- used by --> `savedObjects`. 
We can work around it exposing `setDefaults` via start contract as well, but it's not something we want to support in the long term.